mongo/client_examples_test.go

```diff
@@ -0,0 +1,200 @@
+	// build its view of the deployment.
+	// See https://docs.mongodb.com/manual/reference/connection-string/ for more information about SRV.
+
+	// Versions 1.1.0 or later of the driver support SRV polling for mongos discovery. Prior to this version, using
+	// an SRV connection string for a sharded cluster was supported, but the hosts were only resolved during initial
+	// connection string parsing. Because individual mongos servers are not aware of each other, added/removed mongos
+	// nodes wouldn't be discovered without an application restart. SRV polling allows the driver to periodically
+	// perform DNS resolution in the background and update the its view of the cluster when nodes are added/removed.
+
+	clientOpts := options.Client().ApplyURI("mongodb+srv://mongodb.example.com")
+	client, err := mongo.Connect(context.TODO(), clientOpts)
+	if err != nil {
+		log.Fatal(err)
+	}
+	_ = client
+}
+
+func ExampleConnect_direct() {
+	// Create a direct connection to a host. The driver will send all requests to that host and will not
+	// automatically discover other hosts in the deployment.
+
+	clientOpts := options.Client().ApplyURI("mongodb://localhost:27017/?connect=direct")
+	client, err := mongo.Connect(context.TODO(), clientOpts)
+	if err != nil {
+		log.Fatal(err)
+	}
+	_ = client
+}
+
+func ExampleConnect_sCRAM() {
+	// Configure a Client with SCRAM authentication (https://docs.mongodb.com/manual/core/security-scram/).
+	// The default authentication database for SCRAM is "admin". This can be configured via the
+	// database field in the URI or the AuthSource field in the options.Credential struct.
+	// SCRAM is the default auth mechanism so specifying a mechanism is not required.
+
+	// To configure auth via URI instead of a Credential, use
+	// "mongodb://user:password@localhost:27017".
+	credential := options.Credential{
+		Username: "user",
+		Password: "password",
+	}
+	clientOpts := options.Client().ApplyURI("mongodb://localhost:27017").SetAuth(credential)
+	client, err := mongo.Connect(context.TODO(), clientOpts)
+	if err != nil {
+		log.Fatal(err)
+	}
+	_ = client
+}
+
+func ExampleConnect_x509() {
+	// Configure a Client with X509 authentication (https://docs.mongodb.com/manual/core/security-x.509/).
+
+	// X509 can be configured with different sets of options in the connection string:
+	// 1. tlsCAFile (or SslCertificateAuthorityFile): Path to the file with either a single or bundle of certificate
+	// authorities to be considered trusted when making a TLS connection.
+	// 2. tlsCertificateKeyFile (or SslClientCertificateKeyFile): Path to the client certificate file or the client
+	// private key file. In the case that both are needed, the files should be concatenated.
+
+	// The SetAuth client option should also be used. The username field is optional. If it is not specified, it will
+	// be extracted from the certificate key file. The AuthSource is required to be $external.
+
+	caFilePath := "path/to/cafile"
```

Comment:
Is it a relative path?  If not, prefix with "/" to make it a bit clearer

Reply:
iirc, it can be a relative path.